tables/tests/test_extract.py

```
@@ -0,0 +1,605 @@
+    config = {
+        "extract": {"worksheet_name": "test_worksheet_1", "id_tag": "TESTID12"},
+        "process": {},
+    }
+    return config
+
+
+@pytest.fixture
+def table_with_invalid_end_tag_config():
+    config = {
+        "extract": {"worksheet_name": "test_worksheet_1", "id_tag": "TESTID13"},
+        "process": {},
+    }
+    return config
+
+
+@pytest.fixture
+def table_with_merged_cells_config():
+    """1 merged column and a second column"""
+    config = {
+        "extract": {"worksheet_name": "test_worksheet_1", "id_tag": "TESTID14"},
+        "process": {"merged_header_rows": [0]},
+    }
+    return config
+
+
+def extract_process(table_extractor, config):
+    tables = table_extractor.extract(**config["extract"])
+    processor = TableProcessor(**config["process"])
+    for table in tables:
+        processor.process(table)
+    return tables
+
+
+def test_basic_table_extract_process(table_extractor, basic_table_config):
+    """
+    GIVEN a table schema and a worksheet containing a matching table
+    WHEN an extraction is attempted
+    THEN a single table is returned as expected, with a mapping from
+    """
+    tables = extract_process(table_extractor, basic_table_config)
+    assert len(tables) == 1, f"Exactly one table should be extracted, but {len(tables)} were"
+    table = tables[0]
+    expected_table = pd.DataFrame(
+        data={
+            "StringColumn": ["String1", "String2", "String3"],
+            "IntColumn": ["1", "2", "2"],
+            "DropdownColumn": ["Yes", "No", "Yes"],
+            "UniqueColumn": ["Unique1", "Unique2", "Unique3"],
+        },
+    )
+    assert_frame_equal(table.df, expected_table)
+    assert table.first_row_idx == 2
+    assert table.first_col_idx == 0
+    assert table.col_idx_map == {"DropdownColumn": 2, "IntColumn": 1, "StringColumn": 0, "UniqueColumn": 3}
+
+
+def test_table_extract_and_process_when_no_tables_exist(table_extractor, basic_table_config):
```

Also as mentioned in another comment, I feel that extraction should raise an exception if no tables exist, this seems like something we don't want to silently pass on